pyk/src/pyk/kast.py

```diff
@@ -63,6 +63,17 @@ def _check_node(cls: Type[T], d: Dict[str, Any], expected: Optional[str] = None)
         if actual != expected:
             raise ValueError(f"Expected '{expected}' as 'node' value, found: '{actual}'")
 
+    def _as_shallow_tuple(self) -> Tuple[Any, ...]:
+        # shallow copy version of dataclass.astuple.
+        return tuple([self.__dict__[field.name] for field in fields(type(self))])
+
+    def __lt__(t1, t2):
+        if not isinstance(t2, KAst):
+            return NotImplemented
+        if type(t1) == type(t2):
+            return t1._as_shallow_tuple() < t2._as_shallow_tuple()
+        return type(t1).__name__ < type(t2).__name__
+
```

Comment:
@tothtamas28 I don't use `@dataclass(order=True)` because that doesn't work with ABCs/inheritance.